driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -930,6 +933,214 @@ public Date deserialize(ByteBuffer bytes) {
         }
     }
 
+    static class SimpleDateCodec extends TypeCodec<DateWithoutTime> {
+
+        public static final SimpleDateCodec instance = new SimpleDateCodec();
+        private static final Pattern IS_LONG_PATTERN = Pattern.compile("^-?\\d+$");
+
+        private static final String[] patterns = new String[] {
+                                                              "yyyy-MM-dd",
+                                                              "yyyy-MM-ddZ"
+        };
+
+        private SimpleDateCodec() {}
+
+        private static DateWithoutTime parseDate(String str, final String[] parsePatterns) {
+            SimpleDateFormat parser = new SimpleDateFormat();
+            parser.setLenient(false);
+            parser.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            ParsePosition pos = new ParsePosition(0);
+            for (String parsePattern : parsePatterns) {
+                parser.applyPattern(parsePattern);
+                pos.setIndex(0);
+
+                Date date = parser.parse(str, pos);
+                if (date != null && pos.getIndex() == str.length()) {
+                    return DateWithoutTime.fromMillis(date.getTime());
+                }
+            }
+            throw new IllegalArgumentException("Unable to parse the date: " + str);
+        }
+
+        @Override
+        public DateWithoutTime parse(String value) {
+            if (IS_LONG_PATTERN.matcher(value).matches()) {
+                try {
+                    return DateWithoutTime.fromSimpleDate(Integer.parseInt(value));
+                } catch (NumberFormatException e) {
+                    throw new InvalidTypeException(String.format("Cannot parse date value from \"%s\"", value));
+                }
+            }
+
+            try {
+                return parseDate(value, patterns);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidTypeException(String.format("Cannot parse date value from \"%s\"", value));
+            }
+        }
+
+        @Override
+        public String format(DateWithoutTime value) {
+            return value.toString();
+        }
+
+        @Override
+        public ByteBuffer serialize(DateWithoutTime value) {
+            return IntCodec.instance.serializeNoBoxing(value.getDays());
+        }
+
+        @Override
+        public DateWithoutTime deserialize(ByteBuffer bytes) {
+            return DateWithoutTime.fromSimpleDate(IntCodec.instance.deserializeNoBoxing(bytes));
```

Comment:
OK, I realize now that this is handled in `DateWithoutTime` by adding/subtracting `Integer.MIN_VALUE`, and relying on overflow. Still I'd rather do all these transformations in the codec, and have a "normal" (epoch = 0) `days` field in `DateWithoutTime`.